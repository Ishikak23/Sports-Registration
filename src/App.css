/* *{
  background-image: url("../public/Sport_balls.png");
} */

.App {
  text-align: center;
  width: 100%;
}

.header {
  gap: 10px;
  padding: 10px;
  background-color: #658864;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  width: 100%;
  display: flex;
  justify-content: center;
  overflow: hidden;
}

.body-wrapper {
  position: absolute;
  overflow: hidden;
  margin-top: 100px;
}

.body {
  display: flex;
  flex-direction: row;
  gap: 10px;
  padding: 10px;
  width: 100%;
  background-color: #eeeeee;

  overflow: hidden;
}

.total-event {
  width: 50%;
}

.selected-event {
  width: 50%;
}

.vl {
  border-left: 6px solid green;
  height: 100%;
}

.cardList {
  display: flex;
  flex-direction: row;
  gap: 20px;
  padding: 10px;
  margin: 10px;
  flex-wrap: wrap;
}

.card {
  width: 300px;
  height: 162px;
  /* border: 1px solid black; */
  /* box-shadow: 1px 1px 5px 5px #888888;; */
  display: flex;
  flex-direction: row;
  padding: 10px;
  border-radius: 8px;
  gap: 10px;
  background-color: white;
}

.card > img {
  width: 45%;
}

.cardTitle {
  margin: 8px 25px;
}

.cardDescription {
  margin: 6px 0px;
}

.cardTime {
  margin: 6px 0px;
}

.alert-box {
  width: "30%%";
  position: "fixed";
  top: 20;
  right: 10;
}

.alert-box-conatiner {
  background-color: "#FDEDED";
  display: "flex";
  flex-direction: "row";
  gap: "10px";
  padding: "15px";
  border-radius: "8px";
}

.add-btn {
  border-radius: 4px;
  padding: 5px 10px;
  border: 1px solid rgb(17, 57, 200);
  background-color: rgb(17, 57, 200);
  color: white;
  cursor: pointer;
}

.add-btn :hover {
  border: 1px solid black;
}

.disable-btn {
  border-radius: 4px;
  padding: 5px 10px;
  border: 1px solid rgb(215, 218, 228);
  background-color: rgb(127, 134, 148);
  color: rgb(231, 224, 224);
  cursor: default;
}

.remove-btn {
  border-radius: 4px;
  padding: 5px 10px;
  border: 1px solid rgb(128, 122, 122);
  background-color: rgba(201, 192, 192, 0.97);
  color: black;
  cursor: pointer;
}

.footer {
  gap: 10px;
  padding: 2px;
  background-color: #658864;
  position: fixed;
  bottom: 0;
  width: 100%;
  display: flex;
  justify-content: center;
  overflow: hidden;
}
