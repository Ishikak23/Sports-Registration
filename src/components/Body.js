import Card from "./Card";
import { getMockData } from "../utility";
import { useEffect, useState } from "react";
import { mockSportsData } from "../mockData";
import ShowAlert from "./ShowAlert";
import { checkAvailbleTimeSlot } from "../helper";

const Body = () => {
  const getEventMockData = async () => {
    const response = await getMockData();
    setEventdata(response);
  };
  const [eventData, setEventdata] = useState(mockSportsData);
  const [selectedEventData, setSelectedEventData] = useState([]);
  const [showToast, setShowToast] = useState(false);
  const [toastTitle, setToastTitle] = useState("Event Update Failed!");
  const [toastDescription, setToastDescription] = useState("");

  useEffect(() => {
    getEventMockData();
  }, []);

  const handleSelectedevent = (sport) => {
    if (selectedEventData.length < 3) {
      if (checkAvailbleTimeSlot(sport, selectedEventData)) {
        const selectedEventList = eventData.filter(
          (sportsEvent) => sportsEvent.event_name === sport.event_name
        );
        setSelectedEventData([...selectedEventData, ...selectedEventList]);
      } else {
        setToastTitle("Event Update Failed!");
        setToastDescription("Time slot not avilable");
        setShowToast(true);
        setTimeout(() => {
          setShowToast(false);
        }, 5000);
      }
    } else {
      setToastTitle("Event Update Failed!");
      setToastDescription("Only 3 event can be selected");
      setShowToast(true);
      setTimeout(() => {
        setShowToast(false);
      }, 5000);
    }
  };
  const handleUnSelectedevent = (sport) => {
    const updatedSelectedEvent = selectedEventData.filter(
      (sportsEvent) => sportsEvent.event_name !== sport.event_name
    );
    setSelectedEventData(updatedSelectedEvent);
  };

  return (
    <div data-testid="sport-registration-body" className="body-wrapper">
      <div className="body">
        <div className="total-event">
          <h1 style={{ color: "#4E6C50" }}>Total Events</h1>
          <div className="cardList">
            {eventData?.map((sportsEvent) => {
              return (
                <Card
                  key={sportsEvent?.id}
                  totalevent={sportsEvent}
                  selectedEvent={handleSelectedevent}
                  buttonText="Select"
                  isDisabled={
                    selectedEventData.includes(sportsEvent) ? true : false
                  }
                  isAddButton={true}
                />
              );
            })}
          </div>
        </div>
        <div className="vl"></div>
        <div className="selected-event">
          <h1 style={{ color: "#4E6C50" }}>Selected Events</h1>
          <div className="cardList">
            {selectedEventData?.map((sportsEvent) => {
              return (
                <Card
                  key={sportsEvent?.id}
                  totalevent={sportsEvent}
                  selectedEvent={handleUnSelectedevent}
                  buttonText="Remove"
                  isDisabled={false}
                  isAddButton={false}
                />
              );
            })}
          </div>
        </div>
        {showToast && (
          <ShowAlert title={toastTitle} description={toastDescription} />
        )}
      </div>
    </div>
  );
};

export default Body;
