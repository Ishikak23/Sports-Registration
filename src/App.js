import "./App.css";
import Header from "./components/Header";
import Body from "./components/Body";
import Footer from "./components/Footer"

function App() {
  return (
    <div className="App" data-testid="sport-registration">
      <Header />
      <Body />
      <Footer />
    </div>
  );
}

export default App;
