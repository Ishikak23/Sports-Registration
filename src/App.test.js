/* eslint-disable testing-library/no-debugging-utils */
import { render, screen, fireEvent } from "@testing-library/react";
import App from "./App";
import React from "react";
import Body from "./components/Body";
import { configure, mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

configure({ adapter: new Adapter() });
it("renders react component", () => {
  render(<App />);
  const renderedComponent = screen.getByTestId("sport-registration");
  expect(renderedComponent).toMatchSnapshot();
});

it("renders body component", () => {
  const spy = jest.fn();
  const container = mount(<Body />);
  console.log("container", container.debug());
  const addButton = container.find('[data-testid="add-btn"]');
  console.log("a", addButton.first().debug());
  addButton.first().simulate("click");
  expect(spy.mock.calls.length).toEqual(0);
});
