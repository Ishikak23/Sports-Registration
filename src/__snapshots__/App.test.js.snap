// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders react component 1`] = `
<div
  class="App"
  data-testid="sport-registration"
>
  <div
    class="header"
  >
    <img
      alt="Alt"
      src="Sport_balls.png"
      style="width: 5%;"
    />
    <h1
      style="color: white;"
    >
       Sports Registraion Portal
    </h1>
  </div>
  <div
    class="body-wrapper"
    data-testid="sport-registration-body"
  >
    <div
      class="body"
    >
      <div
        class="total-event"
      >
        <h1
          style="color: rgb(78, 108, 80);"
        >
          Total Events
        </h1>
        <div
          class="cardList"
        >
          <div
            class="card"
          >
            <img
              alt="altrenate-img"
              src="https://thumbs.dreamstime.com/b/olympic-swimming-pool-under-water-background-underwater-swim-157395312.jpg"
            />
            <div
              class="cardLayout"
            >
              <h3
                class="cardTitle"
              >
                Butterfly 100M
              </h3>
              <h4
                class="cardDescription"
              >
                Swimming
              </h4>
              <h4
                class="cardTime"
              >
                01:00 PM-02:00 PM
              </h4>
              <button
                class="add-btn"
                data-testid="add-btn"
              >
                Select
              </button>
            </div>
          </div>
          <div
            class="card"
          >
            <img
              alt="altrenate-img"
              src="https://thumbs.dreamstime.com/b/olympic-swimming-pool-under-water-background-underwater-swim-157395312.jpg"
            />
            <div
              class="cardLayout"
            >
              <h3
                class="cardTitle"
              >
                Backstroke 100M
              </h3>
              <h4
                class="cardDescription"
              >
                Swimming
              </h4>
              <h4
                class="cardTime"
              >
                01:30 PM-02:30 PM
              </h4>
              <button
                class="add-btn"
                data-testid="add-btn"
              >
                Select
              </button>
            </div>
          </div>
          <div
            class="card"
          >
            <img
              alt="altrenate-img"
              src="https://thumbs.dreamstime.com/b/olympic-swimming-pool-under-water-background-underwater-swim-157395312.jpg"
            />
            <div
              class="cardLayout"
            >
              <h3
                class="cardTitle"
              >
                Freestyle 400M
              </h3>
              <h4
                class="cardDescription"
              >
                Swimming
              </h4>
              <h4
                class="cardTime"
              >
                03:00 PM-04:00 PM
              </h4>
              <button
                class="add-btn"
                data-testid="add-btn"
              >
                Select
              </button>
            </div>
          </div>
          <div
            class="card"
          >
            <img
              alt="altrenate-img"
              src="https://thumbs.dreamstime.com/b/olympic-swimming-pool-under-water-background-underwater-swim-157395312.jpg"
            />
            <div
              class="cardLayout"
            >
              <h3
                class="cardTitle"
              >
                High Jump
              </h3>
              <h4
                class="cardDescription"
              >
                Athletics
              </h4>
              <h4
                class="cardTime"
              >
                01:00 PM-02:00 PM
              </h4>
              <button
                class="add-btn"
                data-testid="add-btn"
              >
                Select
              </button>
            </div>
          </div>
          <div
            class="card"
          >
            <img
              alt="altrenate-img"
              src="https://thumbs.dreamstime.com/b/olympic-swimming-pool-under-water-background-underwater-swim-157395312.jpg"
            />
            <div
              class="cardLayout"
            >
              <h3
                class="cardTitle"
              >
                Triple Jump
              </h3>
              <h4
                class="cardDescription"
              >
                Athletics
              </h4>
              <h4
                class="cardTime"
              >
                04:00 PM-05:00 PM
              </h4>
              <button
                class="add-btn"
                data-testid="add-btn"
              >
                Select
              </button>
            </div>
          </div>
          <div
            class="card"
          >
            <img
              alt="altrenate-img"
              src="https://thumbs.dreamstime.com/b/olympic-swimming-pool-under-water-background-underwater-swim-157395312.jpg"
            />
            <div
              class="cardLayout"
            >
              <h3
                class="cardTitle"
              >
                Long Jump
              </h3>
              <h4
                class="cardDescription"
              >
                Athletics
              </h4>
              <h4
                class="cardTime"
              >
                05:00 PM-06:00 PM
              </h4>
              <button
                class="add-btn"
                data-testid="add-btn"
              >
                Select
              </button>
            </div>
          </div>
          <div
            class="card"
          >
            <img
              alt="altrenate-img"
              src="https://thumbs.dreamstime.com/b/olympic-swimming-pool-under-water-background-underwater-swim-157395312.jpg"
            />
            <div
              class="cardLayout"
            >
              <h3
                class="cardTitle"
              >
                100M Sprint
              </h3>
              <h4
                class="cardDescription"
              >
                Athletics
              </h4>
              <h4
                class="cardTime"
              >
                05:00 PM-06:00 PM
              </h4>
              <button
                class="add-btn"
                data-testid="add-btn"
              >
                Select
              </button>
            </div>
          </div>
          <div
            class="card"
          >
            <img
              alt="altrenate-img"
              src="https://thumbs.dreamstime.com/b/olympic-swimming-pool-under-water-background-underwater-swim-157395312.jpg"
            />
            <div
              class="cardLayout"
            >
              <h3
                class="cardTitle"
              >
                Lightweight 60kg
              </h3>
              <h4
                class="cardDescription"
              >
                Boxing
              </h4>
              <h4
                class="cardTime"
              >
                06:00 PM-07:00 PM
              </h4>
              <button
                class="add-btn"
                data-testid="add-btn"
              >
                Select
              </button>
            </div>
          </div>
          <div
            class="card"
          >
            <img
              alt="altrenate-img"
              src="https://thumbs.dreamstime.com/b/olympic-swimming-pool-under-water-background-underwater-swim-157395312.jpg"
            />
            <div
              class="cardLayout"
            >
              <h3
                class="cardTitle"
              >
                Middleweight 75 kg
              </h3>
              <h4
                class="cardDescription"
              >
                Boxing
              </h4>
              <h4
                class="cardTime"
              >
                07:00 PM-08:00 PM
              </h4>
              <button
                class="add-btn"
                data-testid="add-btn"
              >
                Select
              </button>
            </div>
          </div>
          <div
            class="card"
          >
            <img
              alt="altrenate-img"
              src="https://thumbs.dreamstime.com/b/olympic-swimming-pool-under-water-background-underwater-swim-157395312.jpg"
            />
            <div
              class="cardLayout"
            >
              <h3
                class="cardTitle"
              >
                Heavyweight 91kg
              </h3>
              <h4
                class="cardDescription"
              >
                Boxing
              </h4>
              <h4
                class="cardTime"
              >
                08:00 PM-10:00 PM
              </h4>
              <button
                class="add-btn"
                data-testid="add-btn"
              >
                Select
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="vl"
      />
      <div
        class="selected-event"
      >
        <h1
          style="color: rgb(78, 108, 80);"
        >
          Selected Events
        </h1>
        <div
          class="cardList"
        />
      </div>
    </div>
  </div>
  <div
    class="footer"
  >
    <div
      style="color: white;"
    >
       Copyright 
      2023
    </div>
  </div>
</div>
`;
